### ISSUE TITLE
calamari server starts, but the GUI doesn't show up

### ISSUE THREAD

**Issue by XinliGuan** [ceph/calamari#332]
I could start the calamari server successfully, accessing the calamari GUI , it shows that the server is not available.
[root@centos7os calamari-clients-1.2.2]# calamari-ctl initialize
[INFO] Loading configuration..
[INFO] Starting/enabling salt...
[INFO] Starting/enabling postgres...
[INFO] Updating database...
[INFO] Initializing web interface...
[INFO] Starting/enabling services...
[INFO] Restarting services...
[INFO] Complete.
[root@centos7os calamari-clients-1.2.2]#
 [root@centos7os tmp]# ps -eaf | grep calamari*
root     17001 19099  0 15:28 ?        00:00:03 /opt/calamari/venv/bin/python /opt/calamari/venv/bin/cthulhu-manager
postgres 17014 15255  0 15:28 ?        00:00:00 postgres: calamari calamari ::1(34259) idle
postgres 17021 15255  0 15:28 ?        00:00:00 postgres: calamari calamari ::1(34262) idle
root     18403 17256  0 16:04 pts/0    00:00:00 grep --color=auto calamari*
root     19242 19099  0 14:12 ?        00:00:01 /opt/calamari/venv/bin/python /opt/calamari/venv/bin/carbon-cache.py --debug --config /etc/graphite/carbon.conf start
apache   20425 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20426 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20427 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20428 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20429 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20430 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20431 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
apache   20432 20422  0 14:12 ?        00:00:00 calamari-httpd  -DFOREGROUND
[root@centos7os tmp]#

Any suggestion to debug the issue, the guest OS is Centos 7 and build the server code following the instruction in this link:

http://karan-mj.blogspot.com/2014/09/ceph-calamari-survival-guide.html

Build the client code following the instruction:

Note: These instructions are deliberately non-system specific. Please visit the wiki and help us write better more distro and release specific instructions.
• install nodejs, a recent version like 0.10.
• install npm, a recent version (1.4 cause troubles building package, use sudo npm install -g npm@next)
• install grunt-cli globally via npm
• Ruby gem and compass via gem
• make sure your system has a recent gnu compatible make binary
• checkout the code git clone git@github.com:ceph/calamari-clients.git
• cd into calamari-clients
• to build: make dist
This will build all the modules.

In the deployment, copy all the items under the subdir (after gunzip and tar -xvf) to /opt/calamari/webapp/content

drwxr-xr-x. 4 root root 4096 Jul 23 16:59 admin
-rw-r--r--. 1 root root 1398 Jul 23 16:59 bootstrap.md
-rw-r--r--. 1 root root 2222 Jul 23 16:59 calamari-clients.suse.spec
-rw-r--r--. 1 root root 1170 Jul 23 16:59 clients.spec
-rw-r--r--. 1 root root 1646 Jul 23 16:59 COPYING
drwxr-xr-x. 5 root root 4096 Jul 23 16:59 dashboard
-rwxr-xr-x. 1 root root  874 Jul 23 16:59 get-versions.sh
-rw-r--r--. 1 root root 1127 Jul 23 16:59 LICENSE.txt
drwxr-xr-x. 4 root root 4096 Jul 23 16:59 login
-rw-r--r--. 1 root root 4569 Jul 23 16:59 Makefile
-rw-r--r--. 1 root root  910 Jul 23 16:59 Makefile.sub
drwxr-xr-x. 5 root root 4096 Jul 23 16:59 manage
-rw-r--r--. 1 root root  856 Jul 23 16:59 README.md
drwxr-xr-x. 2 root root 4096 Jul 23 16:59 screenshots
-rw-r--r--. 1 root root  815 Jul 23 16:59 ui.build.instructions.md
drwxr-xr-x. 3 root root 4096 Jul 23 16:59 utils
drwxr-xr-x. 9 root root 4096 Jul 23 16:59 vagrant
[root@centos7os calamari-clients-1.2.2]#

![issue](https://cloud.githubusercontent.com/assets/13372247/8883739/c1c3e63c-3216-11e5-842b-921943ac44ab.jpg)

**Comment by GregMeno** [comment]
check logs in /var/log/calamari/ for errors

**Comment by XinliGuan** [comment]
dear Gregory;

The root cause is the build for the client code doesn't work, it only build a tarball, not real build, and 
I follow the instruction to kick out a make build-real, it failed in this error:

In the last step it failed in make build-real with the error message:

> gifsicle@0.1.7 postinstall /temp/calamari-repo/calamari-clients/manage/node_modules/grunt-contrib-imagemin/node_modules/gifsicle
> node index.js

â  pre-build test failed, compiling from source...
stream.js:94
      throw er; // Unhandled stream error in pipe.
            ˆ
Error: invalid tar file
    at Extract.Parse._startEntry (/temp/calamari-repo/calamari-clients/manage/node_modules/grunt-contrib-imagemin/node_modules/gifsicle/node_modules/bin-build/node_modules/download/node_modules/decompress/node_modules/tar/lib/parse.js:145:13)

I try to fix the client build code and then retry.

Thanks a lot;

Charley

**Comment by XinliGuan** [comment]
After fixing the grunt-contrib-imagemin issue (the release should be modified to 0.9.4, the issue fixed in 0.9.2 in file package.jason), I kicking off the build process "make build-real"

The build process went will but, it doesn't create the tarball and the content under the build directorry:

The tail of the build message:


Processing as HTML - dist/index.html
Update the HTML to reference our concat/min/revved script files
Update the HTML with the new css filenames
Update the HTML with the new img filenames
Update the HTML with data-main tags
Update the HTML with the data tags
Update the HTML with background imgs, case there is some inline style
Update the HTML with anchors images
Update the HTML with reference in input

Running "usemin:css" (usemin) task

Processing as CSS - dist/styles/main.css
Update the CSS with new img filenames

Done, without errors.
touch build-stamp
make[1]: Leaving directory `/tmp/calamari-repo/calamari-clients/dashboard'
echo '{ "offline": false, "graphite-host": "/graphite" }' \
                > dashboard/dist/scripts/config.json
[root@acme64 calamari-clients]#


My question how should be verify that the build successfully created the valid client code?

My build subdirectory looks: 

total 76
drwxr-xr-x  8 root root 4096 Jul 27 13:18 admin
-rw-r--r--  1 root root 1398 Jul 23 13:46 bootstrap.md
-rw-r--r--  1 root root 2222 Jul 23 13:46 calamari-clients.suse.spec
-rw-r--r--  1 root root 1170 Jul 23 13:46 clients.spec
-rw-r--r--  1 root root 1646 Jul 23 13:46 COPYING
drwxr-xr-x 10 root root 4096 Jul 27 13:21 dashboard
drwxr-xr-x  2 root root 4096 Jul 23 13:46 debian
-rwxr-xr-x  1 root root  874 Jul 23 13:46 get-versions.sh
-rw-r--r--  1 root root 1127 Jul 23 13:46 LICENSE.txt
drwxr-xr-x  8 root root 4096 Jul 27 13:19 login
-rw-r--r--  1 root root 4569 Jul 23 13:46 Makefile
-rw-r--r--  1 root root  910 Jul 23 13:46 Makefile.sub
drwxr-xr-x  9 root root 4096 Jul 27 13:17 manage
-rw-r--r--  1 root root  856 Jul 23 13:46 README.md
drwxr-xr-x  2 root root 4096 Jul 23 13:46 screenshots
-rw-r--r--  1 root root  815 Jul 23 13:46 ui.build.instructions.md
drwxr-xr-x  3 root root 4096 Jul 23 13:46 utils
drwxr-xr-x  9 root root 4096 Jul 23 13:46 vagrant
[root@acme64 calamari-clients]#

where admin, dashboard, login and manage have been updated.

My question how to create new tarball and only keeps the /content 


P.S.
fixing the issue in package.jason:
    "grunt-contrib-copy": "~0.4.1",
    "grunt-contrib-cssmin": "~0.7.0",
    "grunt-contrib-htmlmin": "~0.1.3",
    "grunt-contrib-imagemin": "~0.9.4",  
    "grunt-contrib-jshint": "~0.7.1",

"grunt-contrib-imagemin": "~0.3.1", modified to "~0.9.4" which could fix the issue :

at Extract.Parse._startEntry (/temp/calamari-repo/calamari-clients/manage/node_modules/grunt-contrib-imagemin/node_modules/gifsicle/node_modules/bin-build/node_modules/download/node_modules/decompress/node_modules/tar/lib/parse.js:145:13)

**Comment by XinliGuan** [comment]
I rebuild the calamari client code and reinitialize the server, but the GUI
doesn't show up:
[root@centos7os calamari-clients-1.2.2]# calamari-ctl initialize
[INFO] Loading configuration..
[INFO] Starting/enabling salt...
[INFO] Starting/enabling postgres...
[INFO] Updating database...
[INFO] Initializing web interface...
[INFO] Starting/enabling services...
[INFO] Restarting services...
[INFO] Complete.
I check the /var/log/calamari, but no idea which log should I enable or to
check?

 [root@centos7os log]# cd calamari/
[root@centos7os calamari]# ls -l
total 368
-rw-r--r--. 1 apache apache      0 Jul 27 03:24 calamari.log
-rw-r--r--. 1 apache apache     20 Jul 24 15:04 calamari.log-20150726.gz
-rw-r--r--. 1 apache apache     20 Jul 26 03:41 calamari.log-20150727.gz
-rw-r--r--. 1 root   root   228674 Jul 27 14:36 cthulhu.log
-rw-r--r--. 1 root   root    89387 Jul 26 03:40 cthulhu.log-20150726.gz
-rw-r--r--. 1 root   root    32484 Jul 27 03:23 cthulhu.log-20150727.gz
-rw-r--r--. 1 root   root        0 Jul 27 03:24 httpd_access.log
-rw-r--r--. 1 root   root       98 Jul 24 16:20
httpd_access.log-20150726.gz
-rw-r--r--. 1 root   root       86 Jul 26 03:41
httpd_access.log-20150727.gz
-rw-r--r--. 1 root   root        0 Jul 27 03:24 httpd_error.log
-rw-r--r--. 1 root   root       20 Jul 24 14:12 httpd_error.log-20150726.gz
-rw-r--r--. 1 root   root       20 Jul 26 03:41 httpd_error.log-20150727.gz

best Regards;

Charley

**Comment by XinliGuan** [comment]
I rebuild the calamari client code and reinitialize the server, but the GUI doesn't show up:
[root@centos7os calamari-clients-1.2.2]# calamari-ctl initialize
[INFO] Loading configuration..
[INFO] Starting/enabling salt...
[INFO] Starting/enabling postgres...
[INFO] Updating database...
[INFO] Initializing web interface...
[INFO] Starting/enabling services...
[INFO] Restarting services...
[INFO] Complete.
I check the /var/log/calamari, but no idea which log should I enable or to check?

 [root@centos7os log]# cd calamari/
[root@centos7os calamari]# ls -l
total 368
-rw-r--r--. 1 apache apache      0 Jul 27 03:24 calamari.log
-rw-r--r--. 1 apache apache     20 Jul 24 15:04 calamari.log-20150726.gz
-rw-r--r--. 1 apache apache     20 Jul 26 03:41 calamari.log-20150727.gz
-rw-r--r--. 1 root   root   228674 Jul 27 14:36 cthulhu.log
-rw-r--r--. 1 root   root    89387 Jul 26 03:40 cthulhu.log-20150726.gz
-rw-r--r--. 1 root   root    32484 Jul 27 03:23 cthulhu.log-20150727.gz
-rw-r--r--. 1 root   root        0 Jul 27 03:24 httpd_access.log
-rw-r--r--. 1 root   root       98 Jul 24 16:20 httpd_access.log-20150726.gz
-rw-r--r--. 1 root   root       86 Jul 26 03:41 httpd_access.log-20150727.gz
-rw-r--r--. 1 root   root        0 Jul 27 03:24 httpd_error.log
-rw-r--r--. 1 root   root       20 Jul 24 14:12 httpd_error.log-20150726.gz
-rw-r--r--. 1 root   root       20 Jul 26 03:41 httpd_error.log-20150727.gz

Best Regards;

Charley

**Comment by XinliGuan** [comment]
The only log available is cthulhu.log
and the message is:
2015-07-27 15:00:14,402 - WARNING - cthulhu.server_monitor.salt Re-opening connection to salt-master
2015-07-27 15:00:14,430 - WARNING - cthulhu.salt Re-opening connection to salt-master

**Comment by GregMeno** [comment]
That is surprising. Is there anything in httpd_access.log ?
What services are running?
sudo supervisorctl status
sudo systemctl status httpd
sudo systemctl status salt-master

**Comment by XinliGuan** [comment]
Gregory;

Thank you very much for your help.

To answer your question.

Is there anything in httpd_access.log?

No. the log is empty
 /var/log/calamari
      -rw-r--r--. 1 apache apache      0 Jul 28 03:19 calamari.log
      -rw-r--r--. 1 apache apache     20 Jul 24 15:04
      calamari.log-20150726.gz
      -rw-r--r--. 1 apache apache     20 Jul 26 03:41
      calamari.log-20150727.gz
      -rw-r--r--. 1 apache apache    185 Jul 27 18:48
      calamari.log-20150728.gz
      -rw-r--r--. 1 root   root   143625 Jul 28 10:22 cthulhu.log
      -rw-r--r--. 1 root   root    89387 Jul 26 03:40
      cthulhu.log-20150726.gz
      -rw-r--r--. 1 root   root    32484 Jul 27 03:23
      cthulhu.log-20150727.gz
      -rw-r--r--. 1 root   root    32692 Jul 28 03:18
      cthulhu.log-20150728.gz
      -rw-r--r--. 1 apache apache      0 Jul 27 18:05 exception.log
      -rw-r--r--. 1 root   root        0 Jul 28 03:19 httpd_access.log
      -rw-r--r--. 1 root   root       98 Jul 24 16:20
      httpd_access.log-20150726.gz
      -rw-r--r--. 1 root   root       86 Jul 26 03:41
      httpd_access.log-20150727.gz
      -rw-r--r--. 1 root   root      248 Jul 27 18:48
      httpd_access.log-20150728.gz
      -rw-r--r--. 1 root   root        0 Jul 28 03:19 httpd_error.log
      -rw-r--r--. 1 root   root       20 Jul 24 14:12
      httpd_error.log-20150726.gz
      -rw-r--r--. 1 root   root       20 Jul 26 03:41
      httpd_error.log-20150727.gz
      -rw-r--r--. 1 root   root      274 Jul 27 18:46
      httpd_error.log-20150728.gz
      -rw-r--r--. 1 apache apache   1179 Jul 27 18:46 info.log

sudo supervisorctl status

   [root@centos7os calamari]# sudo supervisorctl status
   carbon-cache                     RUNNING    pid 19242, uptime 3 days,
   20:12:25
   cthulhu                          RUNNING    pid 21991, uptime 18:47:41
   [root@centos7os calamari]#

sudo systemctl status httpd

[root@centos7os calamari]# sudo systemctl status httpd
      httpd.service - The Apache HTTP Server
         Loaded: loaded (/usr/lib/systemd/system/httpd.service; enabled)
         Active: active (running) since Fri 2015-07-24 16:20:03 EDT; 3 days
      ago
        Process: 19101 ExecStop=/bin/kill -WINCH ${MAINPID} (code=exited,
      status=0/SUCCESS)
        Process: 3720 ExecReload=/usr/sbin/httpd $OPTIONS -k graceful
      (code=exited, status=0/SUCCESS)
       Main PID: 19114 (httpd)
         Status: "Total requests: 0; Current requests/sec: 0; Current
      traffic:   0 B/sec"
         CGroup: /system.slice/httpd.service
                 ââ 3747 calamari-httpd  -DFOREGROUND
                 ââ 3748 calamari-httpd  -DFOREGROUND
                 ââ 3749 calamari-httpd  -DFOREGROUND
                 ââ 3750 calamari-httpd  -DFOREGROUND
                 ââ 3751 calamari-httpd  -DFOREGROUND
                 ââ 3752 calamari-httpd  -DFOREGROUND
                 ââ 3753 calamari-httpd  -DFOREGROUND
                 ââ 3754 calamari-httpd  -DFOREGROUND
                 ââ 3755 /usr/sbin/httpd -DFOREGROUND
                 ââ 3756 /usr/sbin/httpd -DFOREGROUND
                 ââ 3757 /usr/sbin/httpd -DFOREGROUND
                 ââ 3776 /usr/sbin/httpd -DFOREGROUND
                 ââ 3777 /usr/sbin/httpd -DFOREGROUND
                 ââ19114 /usr/sbin/httpd -DFOREGROUND
                 ââ30178 /usr/sbin/httpd -DFOREGROUND

      Jul 26 03:41:04 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:04 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:04 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 27 18:05:30 centos7os httpd[19114]: [Mon Jul 27 17:05:30 CDT
      2015]  buil....
      Jul 27 18:05:30 centos7os httpd[19114]: [Mon Jul 27 17:05:30 CDT
      2015]  comp...e
      Jul 27 18:05:31 centos7os python[30180]: SELinux is
      preventing /usr/sbin/htt....

                                               *****  Plugin restorecon
      (92.2 confi...
      Hint: Some lines were ellipsized, use -l to show in full.
      [root@centos7os calamari]#

sudo systemctl status salt-master

      [root@centos7os calamari]# sudo systemctl status salt-master
      salt-master.service - The Salt Master Server
         Loaded: loaded (/usr/lib/systemd/system/salt-master.service;
      enabled)
         Active: active (running) since Mon 2015-07-27 17:33:26 EDT; 16h
      ago
       Main PID: 28486 (salt-master)
         CGroup: /system.slice/salt-master.service
                 ââ28486 /usr/bin/python /usr/bin/salt-master
                 ââ28493 /usr/bin/python /usr/bin/salt-master
                 ââ28494 /usr/bin/python /usr/bin/salt-master
                 ââ28495 /usr/bin/python /usr/bin/salt-master
                 ââ28520 /usr/bin/python /usr/bin/salt-master
                 ââ28521 /usr/bin/python /usr/bin/salt-master
                 ââ28524 /usr/bin/python /usr/bin/salt-master
                 ââ28525 /usr/bin/python /usr/bin/salt-master
                 ââ28526 /usr/bin/python /usr/bin/salt-master
                 ââ28527 /usr/bin/python /usr/bin/salt-master
                 ââ28528 /usr/bin/python /usr/bin/salt-master
                 ââ28529 /usr/bin/python /usr/bin/salt-master

      Jul 27 17:33:26 centos7os systemd[1]: Starting The Salt Master
      Server...
      Jul 27 17:33:26 centos7os salt-master[28486]: [ERROR   ] ACL user
      wwwrun is not available
      Jul 27 17:33:26 centos7os salt-master[28486]: [ERROR   ] ACL user
      www-data is not available
      Jul 27 17:33:26 centos7os systemd[1]: Started The Salt Master Server.
      [root@centos7os calamari]#

The only issue I could see is the problem in salt server, but we have not
configured the Ceph clusters.

Best Regards;

Charley

**Comment by XinliGuan** [comment]
Gregory;

Our team find that we have not configure the port 80 opened to accept the
request.

Now, after we enable the port 80 working, we hit another issue:

calamari.log

2015-07-28 09:45:21,198 - metric_access - django.request Not
Found: /favicon.ico
2015-07-28 09:45:21,258 - metric_access - django.request Not
Found: /favicon.ico

httpd_access.log:

9.41.89.201 - - [28/Jul/2015:10:45:20 -0400] "GET / HTTP/1.1" 302 -
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /dashboard/ HTTP/1.1" 302
-
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /login/?next=/dashboard/
HTTP/1.1" 403 208
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /favicon.ico HTTP/1.1"
404 85
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /favicon.ico HTTP/1.1"
404 85

httpd_error.log

[Tue Jul 28 10:45:21.090393 2015] [autoindex:error] [pid 3757] [client
9.41.89.201:58517] AH01276: Cannot serve
directory /opt/calamari/webapp/content/login/: No matching DirectoryIndex
(index.html) found, and server-generated directory index forbidden by
Options directive

Our team's debug shows the /opt/calamari/webapp/content/login/app
index.html throws exception


Best Regards;

Charley


From:	Charley Guan/Austin/IBM
To:	ceph/calamari <reply
            +00cc0b57e54abefd9d04feaa0fe02e00d6a0777aa2beb76a92cf0000000111ce714792a169ce05ca344b@reply.github.com>
Cc:	Richard Hill/Raleigh/IBM@IBMUS, William
            Irons/Rochester/IBM@IBMUS, Deborah McLemore/Austin/IBM@IBMUS,
            Dave Bostjancic/Poughkeepsie/IBM@IBMUS
Date:	07/28/2015 09:37 AM
Subject:	Re: [calamari] calamari server starts, but the GUI doesn't show
            up (#332)


Gregory;

Thank you very much for your help.

To answer your question.

Is there anything in httpd_access.log?

No. the log is empty
 /var/log/calamari
      -rw-r--r--. 1 apache apache      0 Jul 28 03:19 calamari.log
      -rw-r--r--. 1 apache apache     20 Jul 24 15:04
      calamari.log-20150726.gz
      -rw-r--r--. 1 apache apache     20 Jul 26 03:41
      calamari.log-20150727.gz
      -rw-r--r--. 1 apache apache    185 Jul 27 18:48
      calamari.log-20150728.gz
      -rw-r--r--. 1 root   root   143625 Jul 28 10:22 cthulhu.log
      -rw-r--r--. 1 root   root    89387 Jul 26 03:40
      cthulhu.log-20150726.gz
      -rw-r--r--. 1 root   root    32484 Jul 27 03:23
      cthulhu.log-20150727.gz
      -rw-r--r--. 1 root   root    32692 Jul 28 03:18
      cthulhu.log-20150728.gz
      -rw-r--r--. 1 apache apache      0 Jul 27 18:05 exception.log
      -rw-r--r--. 1 root   root        0 Jul 28 03:19 httpd_access.log
      -rw-r--r--. 1 root   root       98 Jul 24 16:20
      httpd_access.log-20150726.gz
      -rw-r--r--. 1 root   root       86 Jul 26 03:41
      httpd_access.log-20150727.gz
      -rw-r--r--. 1 root   root      248 Jul 27 18:48
      httpd_access.log-20150728.gz
      -rw-r--r--. 1 root   root        0 Jul 28 03:19 httpd_error.log
      -rw-r--r--. 1 root   root       20 Jul 24 14:12
      httpd_error.log-20150726.gz
      -rw-r--r--. 1 root   root       20 Jul 26 03:41
      httpd_error.log-20150727.gz
      -rw-r--r--. 1 root   root      274 Jul 27 18:46
      httpd_error.log-20150728.gz
      -rw-r--r--. 1 apache apache   1179 Jul 27 18:46 info.log

sudo supervisorctl status

   [root@centos7os calamari]# sudo supervisorctl status
   carbon-cache                     RUNNING    pid 19242, uptime 3 days,
   20:12:25
   cthulhu                          RUNNING    pid 21991, uptime 18:47:41
   [root@centos7os calamari]#

sudo systemctl status httpd

[root@centos7os calamari]# sudo systemctl status httpd
      httpd.service - The Apache HTTP Server
         Loaded: loaded (/usr/lib/systemd/system/httpd.service; enabled)
         Active: active (running) since Fri 2015-07-24 16:20:03 EDT; 3 days
      ago
        Process: 19101 ExecStop=/bin/kill -WINCH ${MAINPID} (code=exited,
      status=0/SUCCESS)
        Process: 3720 ExecReload=/usr/sbin/httpd $OPTIONS -k graceful
      (code=exited, status=0/SUCCESS)
       Main PID: 19114 (httpd)
         Status: "Total requests: 0; Current requests/sec: 0; Current
      traffic:   0 B/sec"
         CGroup: /system.slice/httpd.service
                 ââ 3747 calamari-httpd  -DFOREGROUND
                 ââ 3748 calamari-httpd  -DFOREGROUND
                 ââ 3749 calamari-httpd  -DFOREGROUND
                 ââ 3750 calamari-httpd  -DFOREGROUND
                 ââ 3751 calamari-httpd  -DFOREGROUND
                 ââ 3752 calamari-httpd  -DFOREGROUND
                 ââ 3753 calamari-httpd  -DFOREGROUND
                 ââ 3754 calamari-httpd  -DFOREGROUND
                 ââ 3755 /usr/sbin/httpd -DFOREGROUND
                 ââ 3756 /usr/sbin/httpd -DFOREGROUND
                 ââ 3757 /usr/sbin/httpd -DFOREGROUND
                 ââ 3776 /usr/sbin/httpd -DFOREGROUND
                 ââ 3777 /usr/sbin/httpd -DFOREGROUND
                 ââ19114 /usr/sbin/httpd -DFOREGROUND
                 ââ30178 /usr/sbin/httpd -DFOREGROUND

      Jul 26 03:41:04 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:04 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:04 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 26 03:41:05 centos7os python[3782]: SELinux is
      preventing /usr/sbin/http....

                                              *****  Plugin restorecon
      (92.2 confid...
      Jul 27 18:05:30 centos7os httpd[19114]: [Mon Jul 27 17:05:30 CDT
      2015]  buil....
      Jul 27 18:05:30 centos7os httpd[19114]: [Mon Jul 27 17:05:30 CDT
      2015]  comp...e
      Jul 27 18:05:31 centos7os python[30180]: SELinux is
      preventing /usr/sbin/htt....

                                               *****  Plugin restorecon
      (92.2 confi...
      Hint: Some lines were ellipsized, use -l to show in full.
      [root@centos7os calamari]#

sudo systemctl status salt-master

      [root@centos7os calamari]# sudo systemctl status salt-master
      salt-master.service - The Salt Master Server
         Loaded: loaded (/usr/lib/systemd/system/salt-master.service;
      enabled)
         Active: active (running) since Mon 2015-07-27 17:33:26 EDT; 16h
      ago
       Main PID: 28486 (salt-master)
         CGroup: /system.slice/salt-master.service
                 ââ28486 /usr/bin/python /usr/bin/salt-master
                 ââ28493 /usr/bin/python /usr/bin/salt-master
                 ââ28494 /usr/bin/python /usr/bin/salt-master
                 ââ28495 /usr/bin/python /usr/bin/salt-master
                 ââ28520 /usr/bin/python /usr/bin/salt-master
                 ââ28521 /usr/bin/python /usr/bin/salt-master
                 ââ28524 /usr/bin/python /usr/bin/salt-master
                 ââ28525 /usr/bin/python /usr/bin/salt-master
                 ââ28526 /usr/bin/python /usr/bin/salt-master
                 ââ28527 /usr/bin/python /usr/bin/salt-master
                 ââ28528 /usr/bin/python /usr/bin/salt-master
                 ââ28529 /usr/bin/python /usr/bin/salt-master

      Jul 27 17:33:26 centos7os systemd[1]: Starting The Salt Master
      Server...
      Jul 27 17:33:26 centos7os salt-master[28486]: [ERROR   ] ACL user
      wwwrun is not available
      Jul 27 17:33:26 centos7os salt-master[28486]: [ERROR   ] ACL user
      www-data is not available
      Jul 27 17:33:26 centos7os systemd[1]: Started The Salt Master Server.
      [root@centos7os calamari]#

The only issue I could see is the problem in salt server, but we have not
configured the Ceph clusters.

Best Regards;

Charley

**Comment by GregMeno** [comment]
looks like selinux is preventing calamari from working.
selinux support is still in development.
Try disabling it temporarily to see if that is the issue
sudo setenforce 0

**Comment by XinliGuan** [comment]
Thank you for quick response.

I set the setenforce to 0 and rerun the code, but it failed again:



[root@centos7os ~]# sudo setenforce 0
[root@centos7os ~]#

I have rerun the calamari-ctl initialize

httpd_error.log
[root@centos7os calamari]# cat httpd_error.log
[Tue Jul 28 10:45:21.090393 2015] [autoindex:error] [pid 3757] [client
9.41.89.201:58517] AH01276: Cannot serve
directory /opt/calamari/webapp/content/login/: No matching DirectoryIndex
(index.html) found, and server-generated directory index forbidden by
Options directive
[Tue Jul 28 12:11:20.826288 2015] [autoindex:error] [pid 3776] [client
9.41.89.201:58607] AH01276: Cannot serve
directory /opt/calamari/webapp/content/login/: No matching DirectoryIndex
(index.html) found, and server-generated directory index forbidden by
Options directive
[root@centos7os calamari]#

httpd_access.log
[root@centos7os calamari]# cat httpd_access.log
9.41.89.201 - - [28/Jul/2015:10:45:20 -0400] "GET / HTTP/1.1" 302 -
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /dashboard/ HTTP/1.1" 302
-
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /login/?next=/dashboard/
HTTP/1.1" 403 208
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /favicon.ico HTTP/1.1"
404 85
9.41.89.201 - - [28/Jul/2015:10:45:21 -0400] "GET /favicon.ico HTTP/1.1"
404 85
9.41.89.201 - - [28/Jul/2015:12:11:20 -0400] "GET /dashboard/ HTTP/1.1" 302
-
9.41.89.201 - - [28/Jul/2015:12:11:20 -0400] "GET /login/?next=/dashboard/
HTTP/1.1" 403 208
[root@centos7os calamari]#

**Comment by dmick** [comment]
AH01276 looks like it might be relevant. What were the results of your investigation into your problem?

**Comment by XinliGuan** [comment]
Dan;

To answer your question:
AH01276 looks like it might be relevant. What were the results of your
investigation into your problem?

(2) not finding the index.html ...
With above done I have index.html problem now.

[Mon Jul 27 18:05:32.238952 2015] [autoindex:error] [pid 3756] [client
9.56.37.67:51063]
AH01276:
Cannot serve directory /opt/calamari/webapp/content/login/:
No matching DirectoryIndex (index.html) found,
and server-generated directory index forbidden by Options directive

My research in the calamari login area found this

>>>> It's getting late but maybe we have to register to the
apache /opt/calamari/webapp/content/login/app
>>>>  Ran out of tine to investigate but either:  Maybe build needs to
set /opt/calamari/webapp/content/login/app  to apache or
>>>> maybe we have to set the  /opt/calamari/webapp/content/login/app in
the /opt/calamari/webapp/content/login/ location.

/opt/calamari/webapp/content/login/app





(3) useful notes
============================================

This is the debug info I found so far.  Not sure if salt-minon an d
salt-master are part of basic GUI.  The PDF does suggest looking at
some apache logs.  However in the logs defined below did not see anything.

https://download.inktank.com/docs/Calamari%201.1%20Installation%20Guide.pdf

Check Apache error logs on the Calamari Webapp Host
(/var/log/apache2 or /var/log/httpd,
/var/log/calamari, and /var/log/graphite)

Also /var/log/diamond

**Comment by XinliGuan** [comment]
This is our configuration:

/opt/calamari/webapp/content/admin/app
[root@centos7os app]# ls -l in*
-rw-r--r--. 1 root root 3301 Jul 27 13:56 index.html
[root@centos7os app]#


is there any configuration issue in ndex.html? or the Apach configuration
has issue?


(See attached file: index.html)(See attached file: 404.html)

Best Regards;

Charley


From:	Charley Guan/Austin/IBM
To:	ceph/calamari <reply
            +00cc0b579a31cdbaa1598ffc9c321cfbd34641eae874ca2d92cf0000000111cf6f1792a169ce05ca344b@reply.github.com>
Date:	07/28/2015 11:45 AM
Subject:	Re: [calamari] calamari server starts, but the GUI doesn't show
            up (#332)



Dan;

To answer your question:
AH01276 looks like it might be relevant. What were the results of your
investigation into your problem?

(2) not finding the index.html ...
With above done I have index.html problem now.

[Mon Jul 27 18:05:32.238952 2015] [autoindex:error] [pid 3756] [client
9.56.37.67:51063]
AH01276:
Cannot serve directory /opt/calamari/webapp/content/login/:
No matching DirectoryIndex (index.html) found,
and server-generated directory index forbidden by Options directive

My research in the calamari login area found this

>>>> It's getting late but maybe we have to register to the
apache /opt/calamari/webapp/content/login/app
>>>>  Ran out of tine to investigate but either:  Maybe build needs to
set /opt/calamari/webapp/content/login/app  to apache or
>>>> maybe we have to set the  /opt/calamari/webapp/content/login/app in
the /opt/calamari/webapp/content/login/ location.

/opt/calamari/webapp/content/login/app





(3) useful notes
============================================

This is the debug info I found so far.  Not sure if salt-minon an d
salt-master are part of basic GUI.  The PDF does suggest looking at
some apache logs.  However in the logs defined below did not see anything.

https://download.inktank.com/docs/Calamari%201.1%20Installation%20Guide.pdf

Check Apache error logs on the Calamari Webapp Host
(/var/log/apache2 or /var/log/httpd,
/var/log/calamari, and /var/log/graphite)

Also /var/log/diamond

**Comment by XinliGuan** [comment]
I also reviewed /etc/httpd/conf/httpd.conf and find that
<IfModule dir_module> has been defined.

#
# DirectoryIndex: sets the file that Apache will serve if a directory
# is requested.
#
<IfModule dir_module>
    DirectoryIndex index.html
</IfModule>

**Comment by dmick** [comment]
You really need to read up on Apache configuration; I don't have time to
teach that today.  I don't know for certain what the problem is, but the
hint is in the second half of the message:
"server-generated directory index forbidden by Options directive"

**Comment by XinliGuan** [comment]
Dan, Gregory;

our team has reviewed the Apache installation and find several issues in
the client code deploy.

Our investigation is:


   So what I have done to get this far.

   1. Fixed /etc/httpd/conf.d/calamari.conf  where the
   /opt/calamari/webapp/content/login  ===>
   /opt/calamari/webapp/content/login/app/

   2. Moved the /app files up one level

   /opt/calamari/webapp/content/manage/app/ =>
   /opt/calamari/webapp/content/manage/
   /opt/calamari/webapp/content/dashboard/app/ =>
   /opt/calamari/webapp/content/dashboard/
   /opt/calamari/webapp/content/admin/app/ => /opt/calamari/webapp/content/
   admin/

   I don't know why this is not done like I have it setup above now.  Not
   sure if there is some configuration option when the
   webapp is built so that the ***/app/ directory data is moved up one
   level.    Don't know how the build works for web UI.


We need help to understand why the logout not working:

   Not a web UI Apache guru but it seems that /opt/calamari/webapp/content
   represents the web UI to execute.  If we look in there
   we see the rest_framework, html, css, manage, login, dashboard.  The
   main issue seems to be that Apache wants the index.html
   to be in the root of the folder it needs to look at.  The httpd
   configuration file above indicates calamari root as
   /opt/calamari/webapp/content  and if we look for login, manage or
   dashboard we cannot find the index.html in the
   root of those folders (/opt/calamari/webapp/content/login/
   /opt/calamari/webapp/content/manage and so on).
   By moving the files from the app folder in each
   area /manage  /login /dashboard /admin we can make some things work.

   This seems to imply to me that the /opt/calamari/webapp/content  layout
   is not built right somehow.

As I mentioned in my previous email, I build the client code following the
instruction in this link:


   1.	yum install nodejs npm ruby make git
   2.	npm install -g grunt-cli
   3.	download and install rubygems
   4.	gem install compass
   5.	checkout the code git clone git@github.com:ceph/calamari-clients.git
   6.	cd into calamari-clients folder
   7.	to build: make build-real
   After step 7, all the dist has been created under the
      admin/login/dashboard/manager subdirectories.
   8.   to build the tar ball: make dist

The code structure looks as after we unzip/tar the ball:

[root@centos7os calamari-clients-1.2.2]# ls -l
total 72
drwxr-xr-x. 5 	root root 4096 Jul 27 13:56 admin
-rw-r--r--. 1 	root root 1398 Jul 27 13:56 bootstrap.md
-rw-r--r--. 1 	root root 2222 Jul 27 13:56 calamari-clients.suse.spec
-rw-r--r--. 1 	root root 1170 Jul 27 13:56 clients.spec
-rw-r--r--. 1 	root root 1646 Jul 27 13:56 COPYING
drwxr-xr-x. 7 	root root 4096 Jul 27 13:56 dashboard
-rwxr-xr-x. 1 	root root  874 Jul 27 13:56 get-versions.sh
-rw-r--r--. 1 	root root 1127 Jul 27 13:56 LICENSE.txt
drwxr-xr-x. 5 	root root 4096 Jul 27 13:56 login
-rw-r--r--. 1 	root root 4569 Jul 27 13:56 Makefile
-rw-r--r--. 1 	root root  910 Jul 27 13:56 Makefile.sub
drwxr-xr-x. 6 	root root 4096 Jul 27 13:56 manage
-rw-r--r--. 1 	root root  856 Jul 27 13:56 README.md
drwxr-xr-x. 2 	root root 4096 Jul 27 13:56 screenshots
-rw-r--r--. 1 	root root  815 Jul 27 13:56 ui.build.instructions.md
drwxr-xr-x. 3 	root root 4096 Jul 27 13:56 utils
drwxr-xr-x. 9	root root 4096 Jul 27 13:56 vagrant
[root@centos7os calamari-clients-1.2.2]#

where admin, dashboard, login and manage subdirectories have the app and
dist items.

For example login:
[root@centos7os calamari-clients-1.2.2]# cd login
[root@centos7os login]# ls -l
total 40
drwxr-xr-x. 6 root root  4096 Jul 27 13:56 app
-rw-r--r--. 1 root root   322 Jul 27 13:56 bower.json
-rw-r--r--. 1 root root     0 Jul 27 13:56 build-stamp
drwxr-xr-x. 7 root root  4096 Jul 27 13:56 dist
-rw-r--r--. 1 root root 12351 Jul 27 13:56 Gruntfile.js
-rw-r--r--. 1 root root    24 Jul 27 13:56 Makefile
-rw-r--r--. 1 root root  1066 Jul 27 13:56 package.json
-rw-r--r--. 1 root root   451 Jul 27 13:56 README.md
drwxr-xr-x. 4 root root    44 Jul 27 13:56 test
[root@centos7os login]#

for the app
[root@centos7os app]# ls -l
total 28
-rw-r--r--.  1 root root 4459 Jul 27 13:56 404.html
drwxr-xr-x. 10 root root 4096 Jul 27 13:56 bower_components
-rw-r--r--.  1 root root 4286 Jul 27 13:56 favicon.ico
lrwxrwxrwx.  1 root root   34 Jul 27 13:56 font ->
bower_components/font-awesome/font
drwxr-xr-x.  2 root root   89 Jul 27 13:56 images
-rw-r--r--.  1 root root 3484 Jul 27 13:56 index.html
-rw-r--r--.  1 root root   31 Jul 27 13:56 robots.txt
drwxr-xr-x.  3 root root   59 Jul 27 13:56 scripts
drwxr-xr-x.  2 root root   90 Jul 27 13:56 styles
[root@centos7os app]#

for the dist:

[root@centos7os dist]# ls -l
total 32
-rw-r--r--. 1 root root 4458 Jul 27 13:56 404.html
drwxr-xr-x. 3 root root   22 Jul 27 13:56 bower_components
-rw-r--r--. 1 root root 4286 Jul 27 13:56 favicon.ico
drwxr-xr-x. 2 root root 4096 Jul 27 13:56 font
drwxr-xr-x. 2 root root 4096 Jul 27 13:56 images
-rw-r--r--. 1 root root 3284 Jul 27 13:56 index.html
-rw-r--r--. 1 root root   31 Jul 27 13:56 robots.txt
drwxr-xr-x. 3 root root   42 Jul 27 13:56 scripts
drwxr-xr-x. 2 root root   30 Jul 27 13:56 styles
[root@centos7os dist]#

Our questions

1) In the web site http://ceph.com/category/calamari/

I can not continue the installation steps mentionned in
http://ceph.com/category/calamari/


cd /tmp/opt/calamari/webapp
$ sudo cp -rp content/ /opt/calamari/webapp/content
$ sudo ls -l /opt/calamari/webapp/content


there is no webapp directory


The generated tarball contains the new files, such as dist and app, but we
don't know to deploy them in the /opt/calamari/webapp


2) Do we need to modify any configuration files? such as httd.conf or
calamari.conf?

3) some education questions:

a) does Apache server comes with calamari server code or client code?
b) the webapp/content/app come with calamari configuration or it's standard
httpd setup?
c) How can we uninstall and clean up all the calamari server / client code
and make a fresh deployment? what are the commands to remove calamari?

Best Regards;

Charley

**Comment by XinliGuan** [comment]
Dan, Gregory;

I think that we have messed up with directory structure, and now we pick a
new machine to install both server and client, the server has been
correctly installed, and I checked /opt/calamari/webapp

there is only one subdirector calamari which contains 3 python code.

Is this the correct structure after the server installed?

Installed:
  diamond.noarch 0:3.4.67-0

Complete!
[root@c597n170 src]# cd /opt/calamari/
[root@c597n170 calamari]# cd webapp/
[root@c597n170 webapp]# ls -l
total 0
drwxr-xr-x. 2 apache apache 73 Jul 29 13:04 calamari
[root@c597n170 webapp]# cd calamari/
[root@c597n170 calamari]# ls -l
total 12
-rwxr-xr-x. 1 apache apache 385 Jul 22 17:16 manage.py
-rw-r--r--. 2 apache apache 501 Jul 22 17:19 manage.pyc
-rw-r--r--. 2 apache apache 501 Jul 22 17:19 manage.pyo
[root@c597n170 calamari]#

Best Regards;

Charley

**Comment by XinliGuan** [comment]
Dan, Gregory;

I was talking to Charley on our build issue and reviewed the
http://download.ceph.com/calamari and the
calamari-clients-1.3-rc_12_g7d36e29.el6.x86_64.rpm that has the UI.
Wanted to see what was done by
your team for client install (web ui)

It seems the build package calamari-clients_1.2.2.tar built during calamari
web UI build process has
everything in it.  The proper manage, admin, login and dashboard
directories have a app or dist directory
that seems to represent the web UI and should be copied to
the  /opt/calamari/webapp/content/* location.
All the other stuff in the calamari-clients_1.2.2.tar from the make dist is
extraneous.
                                              
                                              
                                              

Therefore seems like this is the process we need to have to get the web ui
artifacts into the right locaiton
under /opt/calamari/webapp/content/

The process that I think we may have to do
----------------------------------------------------
make build-real
make dist

copy calamari-clients_1.2.2.tar to /tmp/calamari
untar  calamari-clients_1.2.2.tar  in /tmp/calamari

in /opt/calamari/webapp/content
remove  all cotents of manage, dashboard, admin and login
directories manage, dashboard, admin and login now empty
copy in the new web ui build
copy
tmp/calamari-clients_1.2.2/manage/app/*    /opt/calamari/webapp/content/manage
copy
tmp/calamari-clients_1.2.2/admin/app/*     /opt/calamari/webapp/content/admin
copy
tmp/calamari-clients_1.2.2/login/app/*     /opt/calamari/webapp/content/login
copy
tmp/calamari-clients_1.2.2/dashboard/app/* /opt/calamari/webapp/content/dashboard

Doing this gets the index.html into the root webapp/contents/**/ directory.
And no configuration file change is required.

This seems to be step we need to make...

Phone:    External: 845-435-1842   Internal: 8+295-1842
Internet: dbostjan@us.ibm.com

----- Forwarded by Dave Bostjancic/Poughkeepsie/IBM on 07/29/2015 02:38 PM
-----

From:	Charley Guan/Austin/IBM
To:	ceph/calamari <reply
            +00cc0b578dffd277eb27309ad73d57d52a6d3de62500a1bc92cf0000000111cfa1c892a169ce05ca344b@reply.github.com>
Cc:	ceph/calamari <calamari@noreply.github.com>, William
            Irons/Rochester/IBM@IBMUS, Dave
            Bostjancic/Poughkeepsie/IBM@IBMUS
Date:	07/29/2015 11:22 AM
Subject:	Re: [calamari] calamari server starts, but the GUI doesn't show
            up (#332)


Dan, Gregory;

our team has reviewed the Apache installation and find several issues in
the client code deploy.

Our investigation is:


   So what I have done to get this far.

   1. Fixed /etc/httpd/conf.d/calamari.conf  where the
   /opt/calamari/webapp/content/login  ===>
   /opt/calamari/webapp/content/login/app/

   2. Moved the /app files up one level

   /opt/calamari/webapp/content/manage/app/ =>
   /opt/calamari/webapp/content/manage/
   /opt/calamari/webapp/content/dashboard/app/ =>
   /opt/calamari/webapp/content/dashboard/
   /opt/calamari/webapp/content/admin/app/ => /opt/calamari/webapp/content/
   admin/

   I don't know why this is not done like I have it setup above now.  Not
   sure if there is some configuration option when the
   webapp is built so that the ***/app/ directory data is moved up one
   level.    Don't know how the build works for web UI.


We need help to understand why the logout not working:

   Not a web UI Apache guru but it seems that /opt/calamari/webapp/content
   represents the web UI to execute.  If we look in there
   we see the rest_framework, html, css, manage, login, dashboard.  The
   main issue seems to be that Apache wants the index.html
   to be in the root of the folder it needs to look at.  The httpd
   configuration file above indicates calamari root as
   /opt/calamari/webapp/content  and if we look for login, manage or
   dashboard we cannot find the index.html in the
   root of those folders (/opt/calamari/webapp/content/login/
   /opt/calamari/webapp/content/manage and so on).
   By moving the files from the app folder in each
   area /manage  /login /dashboard /admin we can make some things work.

   This seems to imply to me that the /opt/calamari/webapp/content  layout
   is not built right somehow.

As I mentioned in my previous email, I build the client code following the
instruction in this link:


   1.	yum install nodejs npm ruby make git
   2.	npm install -g grunt-cli
   3.	download and install rubygems
   4.	gem install compass
   5.	checkout the code git clone git@github.com:ceph/calamari-clients.git
   6.	cd into calamari-clients folder
   7.	to build: make build-real
   After step 7, all the dist has been created under the
      admin/login/dashboard/manager subdirectories.
   8.   to build the tar ball: make dist

The code structure looks as after we unzip/tar the ball:

[root@centos7os calamari-clients-1.2.2]# ls -l
total 72
drwxr-xr-x. 5 	root root 4096 Jul 27 13:56 admin
-rw-r--r--. 1 	root root 1398 Jul 27 13:56 bootstrap.md
-rw-r--r--. 1 	root root 2222 Jul 27 13:56 calamari-clients.suse.spec
-rw-r--r--. 1 	root root 1170 Jul 27 13:56 clients.spec
-rw-r--r--. 1 	root root 1646 Jul 27 13:56 COPYING
drwxr-xr-x. 7 	root root 4096 Jul 27 13:56 dashboard
-rwxr-xr-x. 1 	root root  874 Jul 27 13:56 get-versions.sh
-rw-r--r--. 1 	root root 1127 Jul 27 13:56 LICENSE.txt
drwxr-xr-x. 5 	root root 4096 Jul 27 13:56 login
-rw-r--r--. 1 	root root 4569 Jul 27 13:56 Makefile
-rw-r--r--. 1 	root root  910 Jul 27 13:56 Makefile.sub
drwxr-xr-x. 6 	root root 4096 Jul 27 13:56 manage
-rw-r--r--. 1 	root root  856 Jul 27 13:56 README.md
drwxr-xr-x. 2 	root root 4096 Jul 27 13:56 screenshots
-rw-r--r--. 1 	root root  815 Jul 27 13:56 ui.build.instructions.md
drwxr-xr-x. 3 	root root 4096 Jul 27 13:56 utils
drwxr-xr-x. 9	root root 4096 Jul 27 13:56 vagrant
[root@centos7os calamari-clients-1.2.2]#

where admin, dashboard, login and manage subdirectories have the app and
dist items.

For example login:
[root@centos7os calamari-clients-1.2.2]# cd login
[root@centos7os login]# ls -l
total 40
drwxr-xr-x. 6 root root  4096 Jul 27 13:56 app
-rw-r--r--. 1 root root   322 Jul 27 13:56 bower.json
-rw-r--r--. 1 root root     0 Jul 27 13:56 build-stamp
drwxr-xr-x. 7 root root  4096 Jul 27 13:56 dist
-rw-r--r--. 1 root root 12351 Jul 27 13:56 Gruntfile.js
-rw-r--r--. 1 root root    24 Jul 27 13:56 Makefile
-rw-r--r--. 1 root root  1066 Jul 27 13:56 package.json
-rw-r--r--. 1 root root   451 Jul 27 13:56 README.md
drwxr-xr-x. 4 root root    44 Jul 27 13:56 test
[root@centos7os login]#

for the app
[root@centos7os app]# ls -l
total 28
-rw-r--r--.  1 root root 4459 Jul 27 13:56 404.html
drwxr-xr-x. 10 root root 4096 Jul 27 13:56 bower_components
-rw-r--r--.  1 root root 4286 Jul 27 13:56 favicon.ico
lrwxrwxrwx.  1 root root   34 Jul 27 13:56 font ->
bower_components/font-awesome/font
drwxr-xr-x.  2 root root   89 Jul 27 13:56 images
-rw-r--r--.  1 root root 3484 Jul 27 13:56 index.html
-rw-r--r--.  1 root root   31 Jul 27 13:56 robots.txt
drwxr-xr-x.  3 root root   59 Jul 27 13:56 scripts
drwxr-xr-x.  2 root root   90 Jul 27 13:56 styles
[root@centos7os app]#

for the dist:

[root@centos7os dist]# ls -l
total 32
-rw-r--r--. 1 root root 4458 Jul 27 13:56 404.html
drwxr-xr-x. 3 root root   22 Jul 27 13:56 bower_components
-rw-r--r--. 1 root root 4286 Jul 27 13:56 favicon.ico
drwxr-xr-x. 2 root root 4096 Jul 27 13:56 font
drwxr-xr-x. 2 root root 4096 Jul 27 13:56 images
-rw-r--r--. 1 root root 3284 Jul 27 13:56 index.html
-rw-r--r--. 1 root root   31 Jul 27 13:56 robots.txt
drwxr-xr-x. 3 root root   42 Jul 27 13:56 scripts
drwxr-xr-x. 2 root root   30 Jul 27 13:56 styles
[root@centos7os dist]#

Our questions

1) In the web site http://ceph.com/category/calamari/

I can not continue the installation steps mentionned in
http://ceph.com/category/calamari/


cd /tmp/opt/calamari/webapp
$ sudo cp -rp content/ /opt/calamari/webapp/content
$ sudo ls -l /opt/calamari/webapp/content


there is no webapp directory


The generated tarball contains the new files, such as dist and app, but we
don't know to deploy them in the /opt/calamari/webapp


2) Do we need to modify any configuration files? such as httd.conf or
calamari.conf?

3) some education questions:

a) does Apache server comes with calamari server code or client code?
b) the webapp/content/app come with calamari configuration or it's standard
httpd setup?
c) How can we uninstall and clean up all the calamari server / client code
and make a fresh deployment? what are the commands to remove calamari?

Best Regards;

Charley

**Comment by dmick** [comment]
"make dist" is for making a source tarball.  That's not very useful for
the clients code.  You want to come up with the
calamari-clients-build-output.tar.gz, or the package (makefile targets
build-real, deb, and rpm).

The instructions that say "make dist" are the final step are incorrect.
I thought those had been fixed long ago.  I will file a PR to fix them.

**Comment by XinliGuan** [comment]
Dan,

As I posted in the help request (issue 97:
https://github.com/ceph/calamari-clients/issues/97 )

we made the make build-real, but what is the next step , should we copy all
the items under /login/app, admin/app, dashboard/app and manage/app to the
clamari master machine under /opt/calamari/webapp/content?


Build target tree					Calamari Master machine

copy tmp/calamari-clients_1.2.2/admin/app/*
/opt/calamari/webapp/content/admin/
copy tmp/calamari-clients_1.2.2/dashboard/app/*
/opt/calamari/webapp/content/dashboard/
copy tmp/calamari-clients_1.2.2/login/app/*
/opt/calamari/webapp/content/login/
copy tmp/calamari-clients_1.2.2/manage/app/*
/opt/calamari/webapp/content/manage/


Doing this gets the index.html into the root webapp/contents/**/ directory.
And no configuration file change is required.



best Regards;

Charley

**Comment by dmick** [comment]
personally, I would build the package.  If you'd rather not install the
package, you can at least then query what it would install.

I *think* make install works too.
